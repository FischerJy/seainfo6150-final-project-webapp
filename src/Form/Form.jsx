import React, { useState } from "react";

const Form = () => {
  const [submittedForm, setSubmittedForm] = useState();
  function onSubmit(e) {
    e.preventDefault();
    const data = new FormData(e.target);
    setSubmittedForm(data);
  }

  return (
    <div className="main">
      {submittedForm ? (
        <div>
          Your form entry was
          <p>email : {submittedForm.get("email")}</p>
          <p>name : {submittedForm.get("name")}</p>
          <p>tel : {submittedForm.get("tel")}</p>
          <p>Favorite development language : {submittedForm.get("language")}</p>
          <p>Favorite sport : {submittedForm.get("sport")}</p>
          <p>weather : {submittedForm.get("weather")}</p>
          <p>info : {submittedForm.get("info")}</p>
        </div>
      ) : (
        <form onSubmit={onSubmit}>
          <p>
            <label>
              Your preferred email (required):
              <input
                type="email"
                name="email"
                placeholder="xxxx@example.com"
                required
              />
            </label>
          </p>
          <p>
            <label>
              Your name (required):
              <input type="text" name="name" required placeholder="name" />
            </label>
          </p>
          <p>
            <label>
              Your tel number (required):
              <input type="tel" placeholder="123-45-678" name="tel" required />
            </label>
          </p>
          <p>
            <label>
              Favorite development language :
              <select name="language">
                <option value="html">HTML</option>
                <option value="css">CSS</option>
                <option value="js">JAVASCRIPT</option>
              </select>
            </label>
          </p>
          <p>
            <label>
              Favorite sport :
              <select name="sport">
                <option value="basketball">BASKETBALL</option>
                <option value="swimming">SWIMMING</option>
                <option value="running">RUNNING</option>
                <option value="dancing">DANCING</option>
              </select>
            </label>
          </p>
          <p>What weather do you like</p>
          <p>
            <label>
              loudy :
              <input type="radio" name="weather" value="loudy" />
            </label>
          </p>
          <p>
            <label>
              rainy :
              <input type="radio" name="weather" value="rainy" />
            </label>
          </p>
          <p>
            <label>
              windy :
              <input type="radio" name="weather" value="windy" />
            </label>
          </p>
          <p>
            <label>
              snowy :
              <input type="radio" name="weather" value="snowy" />
            </label>
          </p>
          <p>
            <label>
              More Info (required, 800 chars max):
              <textarea name="info" maxLength="1000" required />
            </label>
          </p>
          <input type="submit" value="Send it" />
        </form>
      )}
    </div>
  );
};

export default Form;
