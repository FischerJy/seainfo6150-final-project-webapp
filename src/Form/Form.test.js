import React from "react";
import { render } from "@testing-library/react";
import Form from "./Form.jsx";
import { BrowserRouter } from "react-router-dom";



it("renders the Form component correctly", () => {
    const { container } = render(<BrowserRouter>
        <Form />
    </BrowserRouter>);
    expect(container).toMatchSnapshot();
});