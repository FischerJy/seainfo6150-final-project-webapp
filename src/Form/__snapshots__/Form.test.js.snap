// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Form component correctly 1`] = `
<div>
  <div
    class="main"
  >
    <form>
      <p>
        <label>
          Your preferred email (required):
          <input
            name="email"
            placeholder="xxxx@example.com"
            required=""
            type="email"
          />
        </label>
      </p>
      <p>
        <label>
          Your name (required):
          <input
            name="name"
            placeholder="name"
            required=""
            type="text"
          />
        </label>
      </p>
      <p>
        <label>
          Your tel number (required):
          <input
            name="tel"
            placeholder="123-45-678"
            required=""
            type="tel"
          />
        </label>
      </p>
      <p>
        <label>
          Favorite development language :
          <select
            name="language"
          >
            <option
              value="html"
            >
              HTML
            </option>
            <option
              value="css"
            >
              CSS
            </option>
            <option
              value="js"
            >
              JAVASCRIPT
            </option>
          </select>
        </label>
      </p>
      <p>
        <label>
          Favorite sport :
          <select
            name="sport"
          >
            <option
              value="basketball"
            >
              BASKETBALL
            </option>
            <option
              value="swimming"
            >
              SWIMMING
            </option>
            <option
              value="running"
            >
              RUNNING
            </option>
            <option
              value="dancing"
            >
              DANCING
            </option>
          </select>
        </label>
      </p>
      <p>
        What weather do you like
      </p>
      <p>
        <label>
          loudy :
          <input
            name="weather"
            type="radio"
            value="loudy"
          />
        </label>
      </p>
      <p>
        <label>
          rainy :
          <input
            name="weather"
            type="radio"
            value="rainy"
          />
        </label>
      </p>
      <p>
        <label>
          windy :
          <input
            name="weather"
            type="radio"
            value="windy"
          />
        </label>
      </p>
      <p>
        <label>
          snowy :
          <input
            name="weather"
            type="radio"
            value="snowy"
          />
        </label>
      </p>
      <p>
        <label>
          More Info (required, 800 chars max):
          <textarea
            maxlength="1000"
            name="info"
            required=""
          />
        </label>
      </p>
      <input
        type="submit"
        value="Send it"
      />
    </form>
  </div>
</div>
`;
